### PR TITLE
Start chromedriver with PartitionSupervisor

### PR DESCRIPTION
**Pull request by mhanberg**
This will open as many chromedrivers you have as cores, and partition
access to them based on the test pid. This effectively acts as a
"pool" of chromedrivers without needing logic to check them in and out
of the pool.

It just acts as if there is a single chromedriver and it seems to just
work.

- This vendors the PartitionSupervisor from the elixir-lang repository, as it will be released with 1.14 and we won't want to require that immediately.

## TODO

Add the Elixir-lang license to the partition supervisor file.

Related to #365

---

@AndroidOatmeal do you mind testing out this branch?

### CODE REVIEW

**Comment by mhanberg** [comment]
@AndroidOatmeal pinging you again in case the last notification got lost.

**Comment by mhanberg** [comment]
@AndroidOatmeal no worries, I really appreciate it!

**Comment by AndroidOatmeal** [comment]
@mhanberg I spent some time testing this out today and I'm happy to say it seems to work great! I can't tell you how excited I feel lol. Thanks again for the assist here.

I've been running our acceptance test suite over and over with this wallaby branch trying to flush out any hiding issues. Strangely, once I got the usual `(RuntimeError) Wallaby had an internal issue`, but it only happened with a seemingly random pair of tests, and the rest of the suite finished successfully. Even after 10 runs of our suite, I haven't been able to reproduce this again, but I'll let you know if this or other issues pop up.

**Comment by mhanberg** [comment]
Thanks for testing this out!

I just returned from a work trip so I'll hopefully get this tidied up and merged in the next few days. 

I'll probably publish a release candidate and have more people try it out for a little bit then I'll publish the actual release.

**Comment by mhanberg** [comment]
I contracted covid during the aforementioned work trip, so I haven't gotten around to this 😅. I hope to get an RC release out in the next couple days.

**Comment by AndroidOatmeal** [comment]
@mhanberg Sorry to hear about the covid — it really whooped my ass too. Hope you're feeling better!

File: lib/wallaby/partition_supervisor.ex [hunk 1]
```diff
@@ -0,0 +1,565 @@
+#                                 Apache License
```
Comment (mhanberg) [inline lib/wallaby/partition_supervisor.ex:1]: @josevalim is it okay to vendor partition supervisor like this?

Comment (josevalim) [reply lib/wallaby/partition_supervisor.ex:1]: Yes, I believe you need the copyright notice and a link to the repo.

Comment (josevalim) [reply lib/wallaby/partition_supervisor.ex:1]: Here it is: https://github.com/elixir-lang/elixir/blob/main/NOTICE

the second bit.

Comment (mhanberg) [reply lib/wallaby/partition_supervisor.ex:1]: I'll add a link, thank you! I appreciate the clarification.

**Pull request merged**